File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/payload/DeletePostPayload.java
package jp.co.pluswing.mysqlsample.sample.payload;

public class DeletePostPayload {
    private DeletePostPayload(Boolean success) {
        this.success = success;
    }

    public final Boolean success;

    public static DeletePostPayload fail() {
        return new DeletePostPayload(false);
    }
    public static DeletePostPayload success() {
        return new DeletePostPayload(true);
    }
}

File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/payload/DeletePostPayload.kt
package jp.co.pluswing.mysqlsample.sample.payload

data class DeletePostPayload private constructor(val success: Boolean) {
    companion object {
        fun fail(): DeletePostPayload {
            return DeletePostPayload(false)
        }

        fun success(): DeletePostPayload {
            return DeletePostPayload(true)
        }
    }
}